src/mcts/node.h

```diff
@@ -301,9 +309,12 @@ class Node {
   // Index of this node is parent's edge list.
   uint16_t index_;
 
-  // 1 byte fields.
+  // Bit fields using parts of uint8_t fields initialized in the constructor.
   // Whether or not this node end game (with a winning of either sides or draw).
-  Terminal terminal_type_ = Terminal::NonTerminal;
+  Terminal terminal_type_ : 2;
```

Comment:
Could you run a benchmark on a random backend to see how it affects performance?

Reply:
Here's 3 runs with master 1e649cb:
```

Reply:
Here's 3 runs with master 1e649cb with median 100938 nps:
```
./lc0 --smart-pruning-factor=0 --backend=random --backend-opts=uniform=true
ucinewgame
go nodes 2500000

info depth 6 seldepth 10 time 25001 nodes 2500153 score cp 0 nps 100270 tbhits 0 pv b1a3 g8h6 a1b1 b8a6 b1a1 a8b8 a1b1
info depth 6 seldepth 10 time 24834 nodes 2500155 score cp 0 nps 100938 tbhits 0 pv b1a3 g8h6 a1b1 b8a6 b1a1 a8b8 a1b1
info depth 6 seldepth 10 time 24784 nodes 2500153 score cp 0 nps 101138 tbhits 0 pv b1a3 g8h6 a1b1 b8a6 b1a1 a8b8 a1b1
```
And 3 runs with this PR 529406e with median 100872 nps:
```
info depth 6 seldepth 10 time 25160 nodes 2500260 score cp 0 nps 100082 tbhits 0 pv b1a3 b8a6 a1b1 a8b8 b1a1 b8a8 a1b1
info depth 6 seldepth 10 time 24848 nodes 2500114 score cp 0 nps 100872 tbhits 0 pv b1a3 g8h6 a1b1 b8a6 b1a1 a8b8 a1b1
info depth 6 seldepth 10 time 24735 nodes 2500106 score cp 0 nps 101346 tbhits 0 pv b1a3 g8h6 a1b1 b8a6 b1a1 a8b8 a1b1
```

So 99.9% nps of master

Reply:
And using `benchmark --num-positions=6` from #1069:

With master 1e649cb:
```
Total time (ms) : 60013
Nodes searched  : 6023873
Nodes/second    : 100374
```

This PR 529406e:
```
Total time (ms) : 60016
Nodes searched  : 5995393
Nodes/second    : 99895
```
99.5% nps of master

Reply:
Just to document why the bit fields are fast in our usage. The common case behavior of the bit field is to check `Node::IsTerminal() { return terminal_type != Terminal::NonTerminal; }` which is a comparison against `0`. The difference between when terminal_type_ was 1 byte vs 2 bits x86 assembly:
```diff
-        cmpb    $0, 78(%rdi)
+        testb   $3, 78(%rdi)
         setne   %al
```
Where before it would compare the whole byte against 0. With bit fields, it `and`s the value 3 (binary `0b11`) to get the bottom 2 bits for terminal_type_ which uses the same number of instructions when comparing against 0.

For other comparisons not against `0`, e.g., `upper_bound_ == DRAW`:
```
        movb    78(%rdi), %al
        andb    $48, %al
        cmpb    $32, %al
```
It needs to get the desired 2 bits and compare it to the appropriate value. Here `upper_bound_` are bits 4 and 5 (2^4 + 2^5 = 16 + 32 = 48) and DRAW's value is `2` shifted 4 bits left (2 << 4 = 32).

So something maybe to keep in mind as we add more bit fields and optimization.